test/test_autograd.py

```diff
@@ -4192,6 +4192,40 @@ def test_power_function(self):
         c.backward()
         self.assertEqual(b.grad, torch.tensor([-inf, 0., 0.]))
 
+    def test_nanprod_with_nans(self):
```

Comment:
These look good to me but I'll let @albanD check them after the other issues are resolved.